src/extension.js

```diff
@@ -0,0 +1,43 @@
+(function() {
+  class CallbackExtension extends window.Extension {
+    constructor() {
+      super('netatmo-energy-adapter');
+
+      if (!window.Extension.prototype.hasOwnProperty('load')) {
+        this.load();
+      }
+    }
+
+    load() {
+      this.content = '';
+      return fetch(`/extensions/${this.id}/src/views/callback.html`)
+        .then((res) => res.text())
+        .then((text) => {
+          this.content = text;
+        })
+        .catch((e) => console.error('Failed to fetch content:', e));
+    }
+
+    show() {
+      this.view.innerHTML = this.content;
+
+      const queryData = {};
+      const queryParams = new URLSearchParams(window.location.search);
+      for (const [key, value] of queryParams.entries()) {
```

Comment:
isn't this just `Object.fromEntries`? I think this runs client side so it can do even more modern things than the node backend?